### PR TITLE
Fix several hard errors

### PR DESCRIPTION
**Pull request by szepeviktor**
@sandervanhooft Hello! Please consider using @phpstan in CI.

This PR is inspired by `phpstan analyze -l 2 src/`

Level 4 is the minimum for a general project, if you handle money go with Level _max_.

This PR is written blindly, please correct me!

### CODE REVIEW

**Comment by szepeviktor** [comment]
I see why `getResourceCollectionObject()` was left out :)

> Fatal error: Can't inherit abstract function

this may be a PHP 7.2+ feature.

**Comment by szepeviktor** [comment]
Done. I'm lucky :)

The only problem remaining on Level 2:

``` ------ -------------------------------------------------------------------------------------------------------
  Line   Resources/BaseCollection.php
 ------ -------------------------------------------------------------------------------------------------------
  23     Mollie\Api\Resources\BaseCollection::__construct() does not call parent constructor from ArrayObject.
 ------ -------------------------------------------------------------------------------------------------------
```

Please advise.

File: src/Resources/Order.php [hunk 1]
```diff
@@ -158,6 +158,13 @@ class Order extends BaseResource
      */
     public $_links;
 
+    /**
+     * TODO Add description for $_embedded.
```
Comment (sandervanhooft) [inline src/Resources/Order.php]: Needs a description

File: src/Resources/Order.php [hunk 2]
```diff
@@ -274,7 +281,7 @@ public function cancel()
      * @return null
      * @throws \Mollie\Api\Exceptions\ApiException
      */
-    public function cancelLines(array $data)
+    public function cancelLines($data)
```
Comment (sandervanhooft) [inline src/Resources/Order.php]: The `$data` parameter will always have to be an array. (Also check the docblock, `null` is not an option here)

File: src/Endpoints/EndpointAbstract.php [hunk 3]
```diff
@@ -131,34 +132,6 @@ protected function rest_delete($id, array $body = [])
         return ResourceFactory::createFromApiResult($result, $this->getResourceObject());
     }
 
-    /**
```
Comment (sandervanhooft) [inline src/Endpoints/EndpointAbstract.php]: I'd prefer to keep this here (all rest calls together) and include stubs for the required collection calls on this level (`getCollectionResourceName` / `getResourceCollectionObject`).

**Comment by sandervanhooft** [comment]
Thanks @szepeviktor ! Some comments in my review.

**Comment by sandervanhooft** [comment]
Some conflicts due to merge #391

**Comment by szepeviktor** [comment]
I'm glad that you are using @phpstan.
:) I'm not a mollie user! This is an advertisement for PHPStan.

**Comment by szepeviktor** [comment]
@sandervanhooft Although you may pick fixes from this PR.

**Pull request closed without merging**